src/bptl/work_units/valid_sign/tasks.py

```diff
@@ -0,0 +1,283 @@
+
+    * ``signers``: JSON list with signers information. For ValidSign, the first name, the last name and the
+        email address of each signer are required. Example ``signers``:
+
+            .. code-block:: json
+
+                [{
+                    "email": "example.signer@example.com",
+                    "firstName": "ExampleFirstName",
+                    "lastName": "ExampleLastName"
+                },
+                {
+                    "email": "another.signer@example.com",
+                    "firstName": "AnotherFirstName",
+                    "lastName": "AnotherLastName"
+                }]
+
+    * ``packageName``: string. Name of the ValidSign package that contains the documents to sign and the signers.
+        This name appears in the notification-email that is sent to the signers.
+
+    **Sets the process variables**
+
+    * ``packageId``: string. ID of the ValidSign package created by the task.
+    """
+
+    _documents_client = None
+
+    def get_documents_client(self, topic_name: str) -> ZGWClient:
+        if self._documents_client is None:
+            default_services = TaskMapping.objects.get(
+                topic_name=topic_name
+            ).defaultservice_set.select_related("service")
+            services_by_alias = {svc.alias: svc.service for svc in default_services}
+
+            alias = "DocumentenAPI"
+            if alias not in services_by_alias:
+                raise RuntimeError(f"Service alias '{alias}' not found.")
+
+            self._documents_client = services_by_alias[alias].build_client()
+        return self._documents_client
+
+    def format_signers(self, signers: List[dict]) -> List[dict]:
+        """Format the signer information into an array of JSON objects as needed by ValidSign."""
+
+        return [{"type": "SIGNER", "signers": [signer]} for signer in signers]
+
+    def _get_documents_from_api(self) -> List[Tuple[str, bytes]]:
+        """Retrieve the documents and their content from the Documenten API."""
+
+        logger.debug("Retrieving documents from Documenten API")
+
+        variables = self.task.get_variables()
+        document_urls = check_variable(variables, "documents")
+        document_client = self.get_documents_client(topic_name="CreateValidSignPackage")
+
+        documents = []
+        for document_url in document_urls:
+            # Retrieving the document
+            document_data = document_client.retrieve(
+                resource="enkelvoudiginformatieobject", url=document_url,
+            )
+
+            # Retrieving the content of the document
+            # Need use requests directly instead of `document_client.request()` since the response is not in JSON format
+            response = requests.get(
+                document_data["inhoud"], headers=document_client.auth.credentials(),
+            )
+
+            documents.append((document_data["titel"], response.content))
```

Comment:
I am curious what will happen if there are many large files being kept in memory - that's probably going to OOM-kill the container.

We should probably have a treshold of maximum filesize, and above that file size write a temporary file (see https://docs.djangoproject.com/en/3.0/ref/files/uploads/#django.core.files.uploadedfile.TemporaryUploadedFile) and read the file again when we send to ValidSign.

Additionally, there should probably be a global limit (like if you have 1000 docs each 1 MB, that's still 1GB in memory!).